src/client/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -54,11 +57,12 @@ public sealed partial class TokenCache : ITokenCacheInternal
         /// The recommended way to get a cache is by using <see cref="IClientApplicationBase.UserTokenCache"/>
         /// and <c>IConfidentialClientApplication.AppTokenCache</c> once the app is created.
         /// </summary>
-        public TokenCache() : this((IServiceBundle)null)
+        [Obsolete("The recommended way to get a cache is by using IClientApplicationBase.UserTokenCache or IClientApplicationBase.AppTokenCache")]
```

Comment:
>[Obsolete("The recommended w [](start = 8, length = 28)

Will this not be sort of a breaking change?

Reply:
Can we change it now with-out updating the major version?

Reply:
I don't think developers use this ctor. We can remove the Obsolete attribute if you think this will cause breaking changes.

Reply:
I don't think it's a matter of whether I think this is a breaking change or not. I think it is. @jmprieur can we ship with this change with-out bumping the major version?

Reply:
@henrik-me  @bgavrilMS 
I agree that this is a breaking change (as if dvelopers have enabled warning as errors in build, they will be broken).
Now I don't think of any scenario where people would want to use this constructor. This is actually bad if they use it, so they want to know, and the fix is easy enough

I don't feel that I'd want to bump up the major number for this, though as this would provoke some resistance, so I'm fine with taking this theorical breaking change (I hope not many people will be affected, and if they are they have to react anyway), without moving to 5.x

Reply:
Can we please add an aka.ms link to the additional cache page in the wiki or in the ms.docs ?

---
In reply to: [339514356](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1448#discussion_r339514356) [](ancestors = 339514356)

Reply:
@bgavrilMS @jennyf19 @henrik-me, I've created a new aka.ms for this:
https://aka.ms/msal-net-web-token-cache

Reply:
And I've created https://aka.ms/msal-net-app-cache-serialization

Reply:
I'll use yours Jean-Marc, mine points to the wiki, which I've updated.